perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -271,81 +262,238 @@ def PercentileCalculator(numbers):
 
-  list_result = []
-  for k in bucket_list_result:
-    list_result.append(k.name)
+    pass
 
-  return list_result
 
+  @abc.abstractmethod
+  def ListObjects(self, bucket, prefix):
+    """List the objects in a bucket given a prefix.
 
-def DeleteObjects(storage_schema, bucket, objects_to_delete,
-                  host_to_connect=None,
-                  objects_deleted=None):
-  """Delete a bunch of objects.
+    Args:
+      bucket: the name of the bucket.
+      prefix: a prefix to list from.
 
-  Args:
-    storage_schema: The address schema identifying a storage. e.g., "gs"
-    bucket: Name of the bucket.
-    objects_to_delete: A list of names of objects to delete.
-    host_to_connect: An optional endpoint string to connect to.
-    objects_deleted: An optional list to record the objects that have been
-        successfully deleted.
-  """
+    Returns:
+      A list of object names.
+    """
+
+    pass
 
-  for object_name in objects_to_delete:
-    try:
-      if _useBotoApi(storage_schema):
-        object_path = '%s/%s' % (bucket, object_name)
-        object_uri = boto.storage_uri(object_path, storage_schema)
-        if host_to_connect is not None:
-          object_uri.connect(host=host_to_connect)
 
+  @abc.abstractmethod
+  def DeleteObjects(self, bucket, objects_to_delete, objects_deleted=None):
+    """Delete a list of objects.
+
+    Args:
+      bucket: the name of the bucket.
+      objects_to_delete: a list of names of objects to delete.
+      objects_deleted: if given, a list to record the objects that
+        have been successfully deleted.
+    """
+
+    pass
+
+
+  @abc.abstractmethod
+  def WriteObjectFromBuffer(self, bucket, object, stream, size):
+    """Write an object to a bucket.
+
+    Exceptions are propagated to the caller, which can decide whether
+    to tolerate them or not. This function will seek() to the
+    beginning of stream before sending.
+
+    Args:
+      bucket: the name of the bucket to write to.
+      object: the name of the object.
+      stream: a read()-able and seek()-able stream to transfer.
+      size: the number of bytes to transfer.
+
+    Returns:
+      a tuple of (start_time, latency).
+    """
+
+    pass
+
+
+  @abc.abstractmethod
+  def ReadObject(self, bucket, object):
+    """Read an object.
+
+    Exceptions are propagated to the caller, which can decide whether
+    to tolerate them or not.
+
+    Args:
+      bucket: the name of the bucket.
+      object: the name of the object.
+
+    Returns:
+      A tuple of (start_time, latency)
+    """
+
+    pass
+
+
+class AzureBackend(ObjectStorageBackend):
+  def __init__(self):
+    if FLAGS.azure_key is None or FLAGS.azure_account is None:
+      raise ValueError('Must specify azure account and key.')
+    self.blobService = BlobService(FLAGS.azure_account, FLAGS.azure_key)
+
+  def ListObjects(self, bucket, prefix):
+    return [obj.name
+            for obj in self.blobService.list_blobs(bucket, prefix=prefix)]
```

Comment:
I believe it matches the style guide, although I don't know if it's "right" :) What indentation would you expect here?